From b0de71598d0ab7d82e0d5405a0169eeecb3cf791 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 02 May 2019 05:33:41 +0300
Subject: [PATCH] [SW-0] clocks: do per platform handling

In ONIX, and PLDM no PLLs implemented therefore we got some fixed clock
frequency we use across system.

Change-Id: Ib03beeb2c0b7185b7302bbd039223ec7412672dd
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 638c6d0..6e62a98 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -11,6 +11,7 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
 
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
 		{
@@ -121,3 +122,25 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
 }
 
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 1200000000ULL / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
+
diff --git a/gaudi/common.c b/gaudi/common.c
index ba10af4..e7c2324 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -49,28 +49,6 @@
 	return mmPSOC_CPU_PLL_BASE;
 }
 
-u64 hl_pll_get_cpu_cycles_per_usec(void)
-{
-	return 1200000000ULL / USEC_PER_SEC;
-}
-
-void hl_uart_init(void)
-{
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
 int hl_pll_partial_init_get(void)
 {
 	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
diff --git a/gaudi/onix.c b/gaudi/onix.c
index c6a57ec..2e8a818 100644
--- a/gaudi/onix.c
+++ b/gaudi/onix.c
@@ -7,6 +7,9 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
+
+#define SYS_CLOCK_FREQUENCY	35000000ULL
 
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
@@ -17,3 +20,23 @@
 {
 	return 1;
 }
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return SYS_CLOCK_FREQUENCY / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+
+	factor = SYS_CLOCK_FREQUENCY / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index c6a57ec..559ac27 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -7,6 +7,9 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
+
+#define SYS_CLOCK_FREQUENCY	153000ULL
 
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
@@ -17,3 +20,23 @@
 {
 	return 1;
 }
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return SYS_CLOCK_FREQUENCY / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+
+	factor = SYS_CLOCK_FREQUENCY / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
diff --git a/goya/asic.c b/goya/asic.c
index 61e5a5b..b034ada 100644
--- a/goya/asic.c
+++ b/goya/asic.c
@@ -11,6 +11,7 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
 
 static struct hl_pll_cfg goya_pll_cfg[] = {
 	{
@@ -43,3 +44,25 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
 }
 
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 1330000000ULL / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
+
diff --git a/goya/common.c b/goya/common.c
index 6abe158..242b33f 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -176,28 +176,6 @@
 	return mmCPU_PLL_BASE;
 }
 
-u64 hl_pll_get_cpu_cycles_per_usec(void)
-{
-	return 1330000000ULL / USEC_PER_SEC;
-}
-
-void hl_uart_init(void)
-{
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
 int hl_pll_partial_init_get(void)
 {
 	return 0;
diff --git a/goya/onix.c b/goya/onix.c
index c6a57ec..9d12335 100644
--- a/goya/onix.c
+++ b/goya/onix.c
@@ -7,6 +7,9 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
+
+#define SYS_CLOCK_FREQUENCY	50000000ULL
 
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
@@ -17,3 +20,23 @@
 {
 	return 1;
 }
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return SYS_CLOCK_FREQUENCY / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+
+	factor = SYS_CLOCK_FREQUENCY / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
diff --git a/goya/pldm.c b/goya/pldm.c
index c6a57ec..68b6864 100644
--- a/goya/pldm.c
+++ b/goya/pldm.c
@@ -7,6 +7,9 @@
 
 #include <preboot.h>
 #include <pll.h>
+#include <uart.h>
+
+#define SYS_CLOCK_FREQUENCY	150000ULL
 
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
@@ -17,3 +20,23 @@
 {
 	return 1;
 }
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return SYS_CLOCK_FREQUENCY / USEC_PER_SEC;
+}
+
+void hl_uart_init(void)
+{
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+
+	factor = SYS_CLOCK_FREQUENCY / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
